# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1518537741 18000
#      Tue Feb 13 11:02:21 2018 -0500
# Node ID 52ca22ba4474947fa75beee3f4550743cab04e2f
# Parent  d2467914ce331b52e555cd8c8c0bfe802519f39a
include hg ID in displayed version info

* ver.m: Show hg ID along with version in default output.
* version.m: New option, "-hgid".

diff --git a/scripts/miscellaneous/ver.m b/scripts/miscellaneous/ver.m
--- a/scripts/miscellaneous/ver.m
+++ b/scripts/miscellaneous/ver.m
@@ -54,28 +54,30 @@
 
 function retval = ver (package = "")
 
   if (nargin > 1)
     print_usage ();
   endif
 
   if (nargout == 0)
+    hg_id = __octave_config_info__ ("hg_id");
+
     [unm, err] = uname ();
 
     if (err)
       os_string = "unknown";
     else
       os_string = sprintf ("%s %s %s %s",
                            unm.sysname, unm.release, unm.version, unm.machine);
     endif
 
     hbar(1:70) = "-";
     desc = {hbar
-            ["GNU Octave Version: " OCTAVE_VERSION]
+            ["GNU Octave Version: " OCTAVE_VERSION " (hg id: " hg_id ")"]
             ["GNU Octave License: " license]
             ["Operating System: " os_string]
             hbar};
 
     printf ("%s\n", desc{:});
 
     if (isempty (package))
       pkg ("list");
@@ -85,17 +87,18 @@ function retval = ver (package = "")
       pkg ("list", package);
     endif
   else
     ## Get the installed packages
     if (isempty (package))
       lst = pkg ("list");
       ## Start with the version info for Octave
       retval = struct ("Name", "Octave", "Version", version,
-                       "Release", [], "Date", []);
+                       "Release", [],
+                       "Date", __octave_config_info__ ("release_date"));
       for i = 1:numel (lst)
         retval(end+1) = struct ("Name", lst{i}.name, "Version", lst{i}.version,
                                 "Release", [], "Date", lst{i}.date);
       endfor
     elseif (strcmpi (package, "Octave"))
       retval = struct ("Name", "Octave", "Version", version,
                        "Release", [], "Date", []);
     else
diff --git a/scripts/miscellaneous/version.m b/scripts/miscellaneous/version.m
--- a/scripts/miscellaneous/version.m
+++ b/scripts/miscellaneous/version.m
@@ -44,16 +44,19 @@
 ## @item @qcode{"-fftw"}
 ## for version information for the linked @sc{fftw},
 ##
 ## @item @qcode{"-blas"}
 ## for version information for the linked @sc{blas},
 ##
 ## @item @qcode{"-lapack"}
 ## for version information for the linked @sc{lapack}.
+##
+## @item @qcode{"-hgid"}
+## the mercurial ID of the sources used to build Octave.
 ## @end table
 ##
 ## The variant with no input and output argument is an alias for the function
 ## @w{@env{OCTAVE_VERSION}} provided for compatibility.
 ## @seealso{OCTAVE_VERSION, ver}
 ## @end deftypefn
 
 ## Author: jwe
@@ -93,16 +96,18 @@ function [v, d] = version (feature)
           v = "no Java available";
         end_try_catch
       case "-fftw"
         v = __octave_config_info__ ("fftw_version");
       case "-blas"
         v = __blas_version__ ();
       case "-lapack"
         v = __lapack_version__ ();
+      case "-hgid"
+        v = __octave_config_info__ ("hg_id");
       otherwise
         error ("version: invalid FEATURE");
     endswitch
   endif
 
 endfunction
 
 
